spacy_curated_transformers/cli/fill_config_transformer.py

```diff
@@ -0,0 +1,469 @@
+                "provide the '--model-name' and '--model-revision' command-line "
+                "arguments to automatically fill in the loader configuration.",
+                exits=1,
+            )
+        assert loader_config is not None
+
+        model_name = loader_config["name"]
+        model_revision = loader_config.get("revision", "main")
+        msg.info(
+            title="Using loader Hugging Face model name and revision:",
+            text=f"{model_name} ({model_revision})",
+        )
+
+    return model_name, model_revision
+
+
+def _lookup_hf_model_type_for_curated_architecture(
+    msg: Printer, config: Config, transformer_name: str
+) -> str:
+    try:
+        transformer = config["components"][transformer_name]["model"]
+        trf_arch_splits = transformer["@architectures"].split(".")
+        assert len(trf_arch_splits) == 3 and trf_arch_splits[-2].endswith("Transformer")
+        curated_arch = trf_arch_splits[-2]
+    except (KeyError, AssertionError):
+        msg.fail(
+            "Pipeline config does not have a valid model architecture for "
+            f"the '{transformer_name}' Curated Transformer component",
+            exits=1,
+        )
+
+    hf_model_type = CURATED_TRANSFORMER_TO_HF_MODEL_TYPE.get(curated_arch)
+    if hf_model_type is None:
+        msg.fail(
+            "Missing Hugging Face model type for Curated Transformer "
+            f"model architecture '{curated_arch}'",
+            exits=1,
+        )
+    assert hf_model_type is not None
+    return hf_model_type
+
+
+def _validate_hf_model_type(
+    msg: Printer,
+    hf_config: Dict[str, Any],
+    expected_model_type: str,
+    transformer_name: str,
+):
+    incoming_hf_model_type = hf_config["model_type"]
+    if incoming_hf_model_type != expected_model_type:
+        error_msg = (
+            f"Hugging Face model of type '{incoming_hf_model_type}' cannot be loaded into "
+            f"Curated Transformer pipe '{transformer_name}' - "
+        )
+        hf_model_type_to_curated_arch = {
+            v: k for k, v in CURATED_TRANSFORMER_TO_HF_MODEL_TYPE.items()
+        }
+
+        if incoming_hf_model_type not in CURATED_TRANSFORMER_TO_HF_MODEL_TYPE.values():
+            error_msg += (
+                f"It is not supported by `spacy-curated-transformers`. The "
+                f"`{hf_model_type_to_curated_arch[expected_model_type]}` architecture "
+                f"expects a Hugging Face model of type '{expected_model_type}'"
+            )
+            msg.fail(
+                error_msg,
+                exits=1,
+            )
+        else:
+            expected_arch = hf_model_type_to_curated_arch[incoming_hf_model_type]
+            error_msg += (
+                f"Change the 'components.{transformer_name}.model.@architectures' entrypoint "
+                f"to use the '{expected_arch}' architecture."
+            )
+            msg.fail(
+                error_msg,
+                exits=1,
+            )
```

Comment:
I assume we don't really want to provide a default model name for each of the types in `CURATED_TRANSFORMER_TO_HF_MODEL_TYPE`, right? If not, could we print at least an example to showcase to the user what is needed? e.g. store a dict that maps `XlmrTransformer` to `xlm-roberta-base` as possible model name and suggest that?

Reply:
Hmm, I'm a fan of neither as we don't want to hardcode anything that's out of our control (and model/repo names are more likely to change than model types). I've added example model names to the error message, all the same.